RFS/src/main/java/com/rfs/cms/OpenSearchCmsClient.java

```diff
@@ -21,9 +17,14 @@ public OpenSearchCmsClient(OpenSearchClient client) {
     }
 
     @Override
-    public boolean createSnapshotEntry(String snapshotName) {
-        ObjectNode initial = OpenSearchCmsEntry.Snapshot.getInitial(snapshotName);
-        return client.createDocument(CMS_INDEX_NAME, CMS_SNAPSHOT_DOC_ID, initial);
+    public CmsEntry.Snapshot createSnapshotEntry(String snapshotName) {
+        OpenSearchCmsEntry.Snapshot newEntry = OpenSearchCmsEntry.Snapshot.getInitial(snapshotName);
+        boolean createdEntry = client.createDocument(CMS_INDEX_NAME, CMS_SNAPSHOT_DOC_ID, newEntry.toJson());
+        if (createdEntry) {
+            return newEntry;
+        } else {
+            return null;            
+        }
```

Comment:
This is a bad design pattern, callers of `createSnapshotEntry(...)` are forced to check if they got an expected result or not - and they have no information on what/how they should handle this class.

Here are a couple of alternatives that will prevent surprise exceptions in the logic of calling code where a state machine could be busted:

#### Return value is often not used, or its not expected to pass/fail
Return `Optional<CmsEntry.Snapshot>` then the caller will only use the value if present, e.g. `createSnapshotEntry(...).ifPresent(entry -> logger.info("hello entry..."))`

#### Throw specific exception
If the snapshot was already created that could be bad - since there is a specific constraint that RFS only works with a single snapshot.  In this case checked exception `SnapshotAlreadyExistsException` would force downstream callers to handle the case where the snapshot was already created - either by aborting executing, deleting the snapshot and recreating it (recovery), or something else.

Reply:
Note; its also a problem that the CMS entry isn't being returned by `client.createDocument(...)` - since it won't have fields that might only be added after creating such as the server-side timestamp.

Reply:
Took a pass at cleaning this up; still feel like there's work to do in the `OpenSearchClient` class but the rest of it looks better.